adder/adder_test.go

```diff
@@ -105,8 +106,20 @@ func TestAdder_ContextCancelled(t *testing.T) {
 	sth := test.NewShardingTestHelper()
 	defer sth.Clean(t)
 
-	mr, closer := sth.GetRandFileMultiReader(t, 50000) // 50 MB
+	lg, closer := sth.GetRandFileReader(t, 50000) // 50 MB
+	st := sth.GetTreeSerialFile(t)
 	defer closer.Close()
+	defer st.Close()
+
+	slf := files.NewSliceFile([]files.DirEntry{
```

Comment:
This test should not need to adder to be interating over a SliceFile. Can you explain the change?

Reply:
A change in `adder.FromFiles` changed this behaviour slightly:

Before the change in https://github.com/ipfs/ipfs-cluster/pull/613/files?utf8=%E2%9C%93&diff=split#diff-a2c1e35d2137b0f7d1899f06c11d87b6L136 we would:
* Enter into the loop
* Check context  (successfully)
* Call `NextFile` and process it
* Loop again
* Check context (timeout in test), return

After the change:
* Enter into the loop, calling `Next`
* Check context (successfully)
* Process the file
* Exit loop as `Next` returns false (note that we don't check context as we don't re-enter the loop)
* Exit `FromFiles` successfully as nothing else used the context

Strictly speaking old behaviour is more correct, but since we added the file fully in both cases, I'd argue that it makes sense to just return the result instead of timing out since we are already on the return path either way.

I needed to make this test take more than one file in order to force the loop to go around at least twice